integration-tests/src/test/java/org/apache/druid/tests/indexer/ITParallelIndexTest.java

```diff
@@ -39,7 +41,17 @@ public void testIndexData() throws Exception
       doIndexTestTest(
           INDEX_DATASOURCE,
           INDEX_TASK,
-          INDEX_QUERIES_RESOURCE
+          INDEX_QUERIES_RESOURCE,
+          false
+      );
+
+      // Index again, this time only choosing the second data file, and without explicit intervals chosen.
+      // The second datafile covers both day segments, so this should replace them, as reflected in the queries.
+      doIndexTestTest(
+          INDEX_DATASOURCE,
+          REINDEX_TASK,
+          REINDEX_QUERIES_RESOURCE,
+          true
```

Comment:
By any chance, is it possible that this makes the test flaky? It will block while `getSegmentVersions()` returns a single version. What happens if overshadowed segments are removed from `CoordinatorServerView` between retries?

Reply:
Good call. I'll do a more direct check that any new versions have shown up.